script/tool/lib/src/update_excerpts_command.dart

```diff
@@ -23,232 +26,302 @@ class UpdateExcerptsCommand extends PackageLoopingCommand {
+                    break;
+                  case '.cpp':
+                    language = 'c++';
+                    break;
+                  default:
+                    language = extension.substring(1);
+                    break;
+                }
+                final String section = match.namedGroup('section')!;
+                final String plaster = match.namedGroup('plaster') ?? '···';
+                if (!excerptSourceFile.existsSync()) {
+                  reportError(
+                      '${file.path}:$lineNumber: specified file "$excerptPath" (resolved to "${excerptSourceFile.path}") does not exist');
+                } else {
+                  excerpt = _extractExcerpt(excerptSourceFile, section, plaster,
+                      language, reportError);
+                }
+                mode = _ExcerptParseMode.pragma;
+              } else {
+                reportError(
+                    '${file.path}:$lineNumber: $_pragma?> pragma does not match expected syntax or is not alone on the line');
+              }
+            }
+          }
+          output.writeln(line);
+          break;
+        case _ExcerptParseMode.pragma:
+          if (!line.startsWith('```')) {
+            reportError(
+                '${file.path}:$lineNumber: expected code block but did not find one');
+            mode = _ExcerptParseMode.normal;
+          } else {
+            if (line.startsWith('``` ')) {
+              reportError(
+                  '${file.path}:$lineNumber: code block was followed by a space character instead of the language (expected "$language")');
+              mode = _ExcerptParseMode.injecting;
+            } else if (line != '```$language') {
+              reportError(
+                  '${file.path}:$lineNumber: code block has wrong language');
+            }
+            mode = _ExcerptParseMode.injecting;
+          }
+          output.writeln(line);
+          break;
+        case _ExcerptParseMode.injecting:
+          if (line == '```') {
+            if (existingBlock.toString() != excerpt) {
+              detectedChange = true;
+            }
+            output.write(excerpt);
+            output.writeln(line);
+            mode = _ExcerptParseMode.normal;
+            language = null;
+            excerpt = null;
+            existingBlock.clear();
+          } else {
+            existingBlock.writeln(line);
+          }
+          break;
+      }
     }
-    final Set<String> submoduleDependencies = <String>{
-      'code_excerpter',
-      'code_excerpt_updater',
-    };
-    final String relativeRootPath =
-        getRelativePosixPath(repoRoot, from: package.directory);
-    for (final String dependency in submoduleDependencies) {
-      editablePubspec.update(<String>[
-        devDependenciesKey,
-        dependency
-      ], <String, String>{
-        'path': '$relativeRootPath/site-shared/packages/$dependency'
-      });
+    if (detectedChange) {
+      if (errors != null) {
```

Comment:
With errors being non-null, this would be an `isNotEmpty` check, and we would eliminate the possibility that a later change to the logic above would make `errors` non-null but empty and incorrectly report errors here.

Reply:
agreed, fixed